pkg/controller/controller_utils.go

```diff
@@ -356,8 +356,8 @@ func (u *UIDTrackingControllerExpectations) ExpectDeletions(rcKey string, delete
 
 // DeletionObserved records the given deleteKey as a deletion, for the given rc.
 func (u *UIDTrackingControllerExpectations) DeletionObserved(rcKey, deleteKey string) {
-	u.uidStoreLock.Lock()
-	defer u.uidStoreLock.Unlock()
+	u.uidStoreLock.RLock()
+	defer u.uidStoreLock.RUnlock()
```

Comment:
I don't think this one is correct - below you do "Get+Delete" and I think we want to guarantee atomicity of them.
Even though in current implementation of ControllerExpectationsInterface it probably will work, I don't think this microoptimization is worth risking potentially breaking that in the future.

[BTW - i think that moving that expectedUids and klog outside of lock will give us way more]